tools/authors/add.php

```diff
@@ -121,11 +133,9 @@
     if ($author_id) {
         // edit specified author
         // get the values from the database
-        $result = mysqli_query(DPDatabase::get_connection(), "SELECT * FROM authors WHERE author_id = $author_id;");
-        if (!$result) {
-            echo "That author doesn't exist!";
-            exit();
-        }
+        $result = DPDatabase::query(sprintf(
+            "SELECT * FROM authors WHERE author_id = %d;",
+            DPDatabase::escape($author_id)));
```

Comment:
`$author_id` is an int, you don't need to escape it with the `%d`.

Reply:
I was worried that they came from the query, which is user controlled, but I see we run them through get_integer_param which validates the interger-ness of it. I'll all validated integers and remove escaping.

Reply:
That actually doesn't matter though -- sprintf("%d") handles that for us and forces anything to an integer which doesn't need escaping which is why we use it for integers.
```
sprintf("%d", "blah") == 0
```